tests/integration/sharding_tests/test_sharding_backups.py

```diff
@@ -200,3 +216,89 @@ async def test_rotate_backup_password(ops_test: OpsTest) -> None:
                 assert backups == 2
     except RetryError:
         assert backups == 2, "Backup not created after password rotation."
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_restore_backup(ops_test: OpsTest, add_writes_to_db) -> None:
```

Comment:
@MiaAltieri you should have tests for removing/readding relation + deleting the cluster and redeploying a new one, then restoring